pkg/evaluator/builtin.go

```diff
@@ -74,6 +75,25 @@ func lenFunc(args []Value) Value {
 	return newError("'len' takes 1 argument of type 'string', array '[]' or map '{}' not " + args[0].Type().String())
 }
 
+var hasDecl = &parser.FuncDecl{
+	Name: "has",
+	Params: []*parser.Var{
+		{Name: "map", T: parser.ANY_TYPE},
```

Comment:
not sure i understand why this is `ANY_TYPE`. Is it because it is a generic function in a language without generics? If so, perhaps needs a comment, or maybe you could somehow make use of `NONE_TYPE` like you do for empty arrays and maps?

Reply:
I think it needs to be ANY because we can't say map of ANY value (just like in go where you for example can't type assert a slice of Any to a slice of int).
none_type does not work - this is real parameter type checking at the parser level.
The Name "map" is just the variable / parameter name (maybe I should change that).

Reply:
renamed to `m`

Reply:
hmmm. Not sure this will / can work. It is more the other way round: when I pass `{}` to a function that expects `{}int` that will work.

I think it's not really possible to create type: map of anything - or at least it is not possible in go.... maybe i should allow it.

let's chat in person tomorrow.